store/storethehash/storethehash.go

```diff
@@ -545,9 +549,13 @@ func (s *sthStorage) populateMetadata(key []byte, values []indexer.Value) ([]ind
 
 func makeIndexKey(m multihash.Multihash) multihash.Multihash {
 	mhb := []byte(m)
+	decoded, _ := multihash.Decode(mhb)
+	encodedCode := make([]byte, binary.Size(decoded.Code))
+	binary.BigEndian.PutUint64(encodedCode, decoded.Code)
 	var b bytes.Buffer
-	b.Grow(len(mhb) + len(indexKeySuffix))
-	b.Write(mhb)
+	b.Grow(len(decoded.Digest) + binary.Size(decoded.Code) + len(indexKeySuffix))
+	b.Write(decoded.Digest)
+	b.Write(encodedCode)
```

Comment:
Similar logic is needed in `makeMetadataKey` since that suffers from the same problem.